lib/engine/game/base.rb

```diff
@@ -2798,13 +2798,24 @@ def additional_ending_after_text
       end
 
       def action_processed(_action)
-        close_corporations_in_close_cell!
+        return unless close_corporations_in_close_cell!
+
+        # Closing corporations can end up removing the current entity's only
+        # available actions, for example if all they could do was sell shares in
+        # the now closed corporation. If that is the case, skip to the next
+        # entity.
+        return unless active_step&.actions(current_entity)&.empty?
+
+        @round.skip_steps
+        @round.next_entity!
```

Comment:
`next_entity!` is not defined in `round/base.rb`. Can it be added there? The auction based classes are the only ones that don't implement it.

Reply:
Good catch, thanks. Done.